src/Expr.cc

```diff
@@ -3443,6 +3443,8 @@ RecordConstructorExpr::RecordConstructorExpr(RecordTypePtr known_rt, ListExprPtr
 				ExprError(err.c_str());
 				}
 			}
+		else if ( known_rt->IsFieldDeprecated(i) )
+			reporter->Warning("%s", known_rt->GetFieldDeprecationWarning(i, false).c_str());
```

Comment:
Thank you! How about we switch all the `reporter->Warning()` calls after `IsFieldDeprecated()` over to `Warn()`, as you do below? The various combos of manual location push/pop and reporter invocations in this file are pretty subtle, and that would make it a little more consistent. Plus, the extra detail in the deprection output seems useful.

Reply:
It's fine!